src/basic/process-util.c

```diff
@@ -275,21 +276,59 @@ int get_process_cmdline(pid_t pid, size_t max_length, bool comm_fallback, char *
 }
 
 void rename_process(const char name[8]) {
+        static char *mm = NULL;
+        static size_t mm_size = 0;
+        size_t l;
+        char *old_mm = NULL;
+        size_t old_mm_size = 0;
+
+        /* This is a like a poor man's setproctitle(). It changes the comm field, argv[0], and also the glibc's
+         * internally used name of the process. For the first one a limit of 16 chars applies; to the second one in
+         * many cases one of 10 (i.e. length of "/sbin/init") — however if we have CAP_SYS_RESOURCES it is unbounded;
```

Comment:
yeah, we do. because of the documented limits in most cases.

the primary reason I much prefer the PR_SET_MM_ARG_START thing over all other solutions is that we actually set up the argv[] array the kernel maintains for us in the exact right way, and not longer than necessary. Normally, rename_process() actually leaves tons of NUL bytes at the end, as we cannot truncate things, but with PR_SET_MM_ARG_END we can do that properly.